budgetApp1/budgetApp1/BudgetViewController.swift

```diff
@@ -9,49 +9,41 @@ import UIKit
 
 class BudgetViewController: UIViewController {
     
+    // MARK: - IBOutlets
+    @IBOutlet private weak var budgetNameLabel: UILabel!
+    @IBOutlet private weak var endDateLabel: UILabel!
+    @IBOutlet private weak var startDateLabel: UILabel!
+    
     private var viewModel: BudgetViewModel? {
         didSet {
-            fillUI()
+            viewModel?.fetchBudgets()
         }
     }
     
-    @IBOutlet weak var budgetNameLabel: UILabel!
-    @IBOutlet weak var endDateLabel: UILabel!
-    @IBOutlet weak var startDateLabel: UILabel!
-    
+    // MARK: - functions
     override func viewDidLoad() {
         super.viewDidLoad()
-        
         viewModel = BudgetViewModel(repository: BudgetRepository(), delegate: self)
-        viewModel?.getBudgets()
-        // Do any additional setup after loading the view.
-         
     }
 
-    private func fillUI() {
-        if !isViewLoaded {
-            return
-        }
-        
-        guard let viewModel = viewModel, let budgetList = viewModel.budgetList, !budgetList.isEmpty else {
-            return
-        }
-        budgetNameLabel.text = budgetList.first?.name
-        startDateLabel.text = budgetList.first?.firstMonth
-        endDateLabel.text = budgetList.first?.lastMonth
+    private func updateView(withBudget budget: Budget) {
+        budgetNameLabel.text = budget.name
+        startDateLabel.text = budget.firstMonth
+        endDateLabel.text = budget.lastMonth
     }
-
 }
 
 // MARK: - ViewModel Delegate
-
 extension BudgetViewController: ViewModelDelegate {
     
     func reloadView() {
-        fillUI()
+        guard let budget = viewModel?.budgetList?.first else {
+            return
+        }
```

Comment:
Please move the return into the same line
Due to the small else block this is a bit cleaner and saves lines

```suggestion
        guard let budget = viewModel?.budgetList?.first else { return }
```

Reply:
Done.